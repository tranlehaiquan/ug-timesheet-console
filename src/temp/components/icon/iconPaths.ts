import actions from "./icons/actions.svg";
import activity from "./icons/activity.svg";
import arrowLeft from "./icons/arrow-left.svg";
import arrowUp from "./icons/arrow-up.svg";
import arrowDown from "./icons/arrow-down.svg";
import attachments from "./icons/attachments.svg";
import attribute from "./icons/attribute.svg";
import attributeFilled from "./icons/attribute-filled.svg";
import availability from "./icons/availability.svg";
import bell from "./icons/bell.svg";
import bellAlert from "./icons/bell-alert.svg";
import bookingGrid from "./icons/booking-grid.svg";
import bookmark from "./icons/bookmark.svg";
import calendar from "./icons/calendar.svg";
import calendarNavRight from "./icons/calendar-nav-right.svg";
import calendarNavLeft from "./icons/calendar-nav-left.svg";
import cancel from "./icons/cancel.svg";
import cardSize from "./icons/card-size.svg";
import chat from "./icons/chat.svg";
import chevronUp from "./icons/chevron-up.svg";
import chevronDown from "./icons/chevron-down.svg";
import chevronLeft from "./icons/chevron-left.svg";
import chevronRight from "./icons/chevron-right.svg";
import contact from "./icons/contact.svg";
import copy from "./icons/copy.svg";
import critical from "./icons/critical.svg";
import dashboard from "./icons/dashboard.svg";
import dateRange from "./icons/date-range.svg";
import details from "./icons/details.svg";
import dispatch from "./icons/dispatch.svg";
import edit from "./icons/edit.svg";
import ellipsisVertical from "./icons/ellipsis-vertical.svg";
import exclamation from "./icons/exclamation.svg";
import filter from "./icons/filter.svg";
import filterNew from "./icons/filter-new.svg";
import glasses from "./icons/glasses.svg";
import help from "./icons/help.svg";
import hide from "./icons/hide.svg";
import info from "./icons/info.svg";
import infoFill from "./icons/info-fill.svg";
import jobFields from "./icons/job-fields.svg";
import jobs from "./icons/jobs.svg";
import location from "./icons/location.svg";
import locked from "./icons/locked.svg";
import logout from "./icons/logout.svg";
import map from "./icons/map.svg";
import mapResource from "./icons/map-resource.svg";
import megaphone from "./icons/megaphone.svg";
import minus from "./icons/minus.svg";
import normal from "./icons/normal.svg";
import notes from "./icons/notes.svg";
import pin from "./icons/pin.svg";
import plus from "./icons/plus.svg";
import remove from "./icons/remove.svg";
import restore from "./icons/restore.svg";
import salesforce from "./icons/salesforce.svg";
import search from "./icons/search.svg";
import settings from "./icons/settings.svg";
import skedulo from "./icons/skedulo.svg";
import skeduloText from "./icons/skedulo-text.svg";
import swimlane from "./icons/swimlane.svg";
import swimlaneVertical from "./icons/swimlane-vertical.svg";
import tickCircle from "./icons/tick-circle.svg";
import tick from "./icons/tick.svg";
import time from "./icons/time.svg";
import trash from "./icons/trash.svg";
import view from "./icons/view.svg";
import warning from "./icons/warning.svg";
import notify from "./icons/notify.svg";
import openLink from "./icons/open-link.svg";
import optimise from "./icons/optimise.svg";
import optimiseFill from "./icons/optimise-fill.svg";
import phone from "./icons/phone.svg";
import planeLanding from "./icons/plane-landing.svg";
import planeLandingGround from "./icons/plane-landing-with-ground.svg";
import planeTakeoff from "./icons/plane-take-off.svg";
import push from "./icons/push.svg";
import recurring from "./icons/recurring.svg";
import reorder from "./icons/reorder.svg";
import resource from "./icons/resource.svg";
import resourceAbove from "./icons/resource-above.svg";
import resourceBelow from "./icons/resource-below.svg";
import resourceAdd from "./icons/resource-add.svg";
import resourceRemove from "./icons/resource-remove.svg";
import scheduling from "./icons/scheduling.svg";
import shift from "./icons/shift.svg";
import shiftOvernight from "./icons/shift-overnight.svg";
import sms from "./icons/sms.svg";
import sortAscending from "./icons/sort-ascending.svg";
import sortDescending from "./icons/sort-descending.svg";
import suggest from "./icons/suggest.svg";
import tasks from "./icons/tasks.svg";
import today from "./icons/today.svg";
import unlocked from "./icons/unlocked.svg";
import unschedule from "./icons/unschedule.svg";
import upDown from "./icons/up-down.svg";
import upload from "./icons/upload.svg";
import urgent from "./icons/urgent.svg";
import warningFill from "./icons/warning-fill.svg";
import zoomIn from "./icons/zoom-in.svg";
import zoomOut from "./icons/zoom-out.svg";

export default {
  actions,
  activity,
  arrowLeft,
  arrowUp,
  arrowDown,
  attachments,
  attribute,
  attributeFilled,
  availability,
  bell,
  bellAlert,
  bookingGrid,
  bookmark,
  calendar,
  calendarNavRight,
  calendarNavLeft,
  cancel,
  cardSize,
  chat,
  chevronUp,
  chevronDown,
  chevronLeft,
  chevronRight,
  contact,
  copy,
  critical,
  dashboard,
  dateRange,
  details,
  dispatch,
  edit,
  ellipsisVertical,
  exclamation,
  filter,
  filterNew,
  glasses,
  help,
  hide,
  info,
  infoFill,
  jobFields,
  jobs,
  location,
  locked,
  logout,
  map,
  mapResource,
  megaphone,
  minus,
  normal,
  notes,
  notify,
  openLink,
  optimise,
  optimiseFill,
  phone,
  pin,
  planeLanding,
  planeLandingGround,
  planeTakeoff,
  plus,
  push,
  recurring,
  remove,
  reorder,
  resource,
  resourceAbove,
  resourceAdd,
  resourceBelow,
  resourceRemove,
  restore,
  salesforce,
  scheduling,
  search,
  settings,
  shift,
  shiftOvernight,
  skedulo,
  skeduloText,
  sms,
  sortAscending,
  sortDescending,
  suggest,
  swimlane,
  swimlaneVertical,
  tasks,
  tick,
  tickCircle,
  time,
  today,
  trash,
  unlocked,
  unschedule,
  upDown,
  upload,
  urgent,
  view,
  warning,
  warningFill,
  zoomIn,
  zoomOut,
};
