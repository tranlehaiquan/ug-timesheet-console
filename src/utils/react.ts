import * as React from 'react'

type IdentityFunction = <T>(fn: T) => T

export const typedMemo: IdentityFunction = React.memo

export const getDisplayName = <T extends any = any>(WrappedComponent: React.ComponentType<T>): string => {
  return WrappedComponent.displayName || WrappedComponent.name || 'Component'
}
