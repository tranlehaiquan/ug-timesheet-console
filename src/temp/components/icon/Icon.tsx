import * as React from "react";
import icons from "./iconPaths";
import classNames from "classnames";

export type IconNames = keyof typeof icons;

interface IProps extends React.SVGAttributes<SVGElement> {
  /**
   * Name of the icon
   */
  name: IconNames;
  /**
   * className applied to the containing div
   */
  className?: string;
  /**
   * Size of the icon to override the default (pixels)
   */
  size?: number;
  /**
   * Function to call on click
   */
  onClick?: (event?: React.MouseEvent<SVGElement>) => void;
}

const DEFAULT_SIZE = 18;

/**
 * The Icon component should be used whenever an icon needs to be displayed. It will naturally inherit any text colour but this can be changed by supplied a class directly. Default size of 18px/18px.
 */
export const Icon = ({
  name,
  className,
  size,
  onClick,
  ...otherProps
}: IProps) => {
  const Component = icons[name];
  return (
    <Component
      {...otherProps}
      height={size || DEFAULT_SIZE}
      width={size || DEFAULT_SIZE}
      onClick={onClick}
      className={classNames("sk-flex-shrink-0 sk-fill-current", className)}
    />
  );
};
