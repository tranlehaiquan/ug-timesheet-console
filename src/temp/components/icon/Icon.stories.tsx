import * as React from 'react'
import { storiesOf } from '@storybook/react'
import { select, number } from '@storybook/addon-knobs'
import { action } from '@storybook/addon-actions'

import { Icon, IconNames } from './Icon'
import icons from './iconPaths'

const iconNames = Object.keys(icons) as IconNames[]

storiesOf('Icons', module)
  .add('Overview', () => (
    <div className="sk-p-4">
      <h3 className="sk-my-4 sk-ml-2 sk-text-blue">{ iconNames.length } icons</h3>
      <div className="sk-flex sk-flex-wrap sk-items-center">
        { iconNames.map((icon, index) => {
          return (
            <div className="sk-w-32 sk-h-32 sk-m-1 sk-p-4 sk-border-3 sk-border-grey-lighter sk-flex sk-flex-col sk-justify-center sk-items-center sk-text-grey" key={ `${name}-${index}` }>
              <Icon name={ icon } size={ 40 } />
              <span className="sk-pt-4 sk-text-sm sk-text-grey">{ icon }</span>
            </div>
          )
        }) }
      </div>
    </div>
  ))
  .add(
    'Icon',
    () => <Icon name={ select('Icon name', iconNames, iconNames[0]) } size={ number('Size', 18) } onClick={ action() } />
  )
