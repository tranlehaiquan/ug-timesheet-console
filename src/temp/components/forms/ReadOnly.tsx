import * as React from "react";
import classnames from "classnames";

import { Icon } from "../icon/Icon";

interface IProps {
  value: string;
  name: string;
  locked?: boolean;
  className?: string;
  placeholderValue?: string;
}

export const ReadOnly: React.SFC<IProps> = ({
  value,
  name,
  placeholderValue,
  locked,
  className,
  ...otherProps
}) => {
  const lockedClasses = "sk-cursor-not-allowed";
  const defaultClasses = "hover:sk-bg-grey-lightest";
  const classes = classnames(
    "sked-read-only group",
    {
      "sk-text-grey": !value,
      [defaultClasses]: !locked,
      [lockedClasses]: locked,
    },
    className
  );
  const iconClasses =
    "sk-flex-shrink-0 sk-invisible group-hover:sk-fill-navy group-hover:sk-visible group-hover:sk-text-grey sk-mx-2 sk-pointer-events-none";

  return (
    <div className="sk-flex">
      <div
        className={classes}
        data-sk-input-id={name}
        data-read-only={!locked}
        {...otherProps}
      >
        {value || placeholderValue || "None"}{" "}
        <Icon
          name={locked ? "locked" : "edit"}
          size={14}
          className={iconClasses}
        />
      </div>
    </div>
  );
};
