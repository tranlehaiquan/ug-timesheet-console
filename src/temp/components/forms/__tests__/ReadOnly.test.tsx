import * as React from "react";
import { shallow } from "enzyme";

import { ReadOnly } from "../ReadOnly";
import { Icon } from "../../icon/Icon";

describe("ReadOnly", () => {
  test("renders correctly", () => {
    // Arrange
    const component = shallow(
      <ReadOnly value="Hello world" name="hello-world" />
    );

    // Assert
    expect(component).toMatchSnapshot();
  });

  test("renders the correct icon on hover", () => {
    // Arrange
    const component = shallow(
      <ReadOnly value="Hello world" name="hello-world" />
    );
    const lockedComponent = shallow(
      <ReadOnly value="Hello world" name="hello-world" locked />
    );

    // Act
    component.simulate("mouseover");

    // Assert
    // Edit
    expect(component.find(Icon).length).toBe(1);
    expect(component.find(Icon).props().name).toBe("edit");

    // Locked
    expect(lockedComponent.find(Icon).length).toBe(1);
    expect(lockedComponent.find(Icon).props().name).toBe("locked");
  });

  test("renders correct default value", () => {
    // Arrange
    const component = shallow(
      <ReadOnly value="" name="hello-world" data-sk-name="sked-readonly" />
    );
    const componentWithPlaceholder = shallow(
      <ReadOnly
        value=""
        placeholderValue="Add something"
        name="hello-world"
        data-sk-name="sked-readonly"
      />
    );
    const componentWithInitialValue = shallow(
      <ReadOnly value="Hello" name="hello-world" data-sk-name="sked-readonly" />
    );

    // Assert
    expect(component.find('[data-sk-name="sked-readonly"]').text()).toBe(
      "None <Icon />"
    );
    expect(
      componentWithPlaceholder.find('[data-sk-name="sked-readonly"]').text()
    ).toBe("Add something <Icon />");
    expect(
      componentWithInitialValue.find('[data-sk-name="sked-readonly"]').text()
    ).toBe("Hello <Icon />");
  });
});
