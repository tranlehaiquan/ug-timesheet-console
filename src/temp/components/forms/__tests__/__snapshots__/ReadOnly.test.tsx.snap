// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadOnly renders correctly 1`] = `
<div
  className="sk-flex"
>
  <div
    className="sked-read-only group hover:sk-bg-grey-lightest"
    data-read-only={true}
    data-sk-input-id="hello-world"
  >
    Hello world
     
    <Icon
      className="sk-flex-shrink-0 sk-invisible group-hover:sk-fill-navy group-hover:sk-visible group-hover:sk-text-grey sk-mx-2 sk-pointer-events-none"
      name="edit"
      size={14}
    />
  </div>
</div>
`;
